### PR TITLE
Remove double delete of PrepareChunk lua callback

### LINKED ISSUE #3598: API: cWorld:PrepareChunk passing nil instead of a function will crash server
Server OS:        Windows
Commit id:        83fa6cb851536839ebd85b1e91fda928f47eb674

### Actual behavior
Calling this function crashes the server:
`cRoot:Get():GetDefaultWorld():PrepareChunk(1, 1, nil)`

### Steps to reproduce the behavior
Run the code from a command handler:
`cRoot:Get():GetDefaultWorld():PrepareChunk(1, 1, nil)`

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #3598

### CODE REVIEW

**Comment by Seadragon91** [comment]
Tested and works fine.
Found out that without that fix passing a valid callback also crashes the server:
```
cRoot:Get():GetDefaultWorld():PrepareChunk(1, 1,
	function(ChunkX, ChunkZ)
		LOG("Chunk done.")
	end)
```

**Comment by Seadragon91** [comment]
Okay please do not merge... Found a problem.

**Comment by Seadragon91** [comment]
It works "fine" under windows, but crashes under linux. Doesn't matter if a valid or not a valid callback is passed.

**Comment by peterbell10** [comment]
Can you give any more information about the crash?

**Comment by Seadragon91** [comment]
Okay sorry, did something wrong. No memory leak with valgrind and no crash in release mode.

I get an assertion error in debug mode, if the code is called multiple times:
```
[9760e6ec02d9a174|22:39:58] Assertion failed: cCanonLuaStates::GetCanonState(m_LuaState)->m_CS.IsLockedByCurrentThread(), file /home/lukas/cpp/cuberite/src/Bindings/LuaState.cpp, line 2402
./Cuberite_debug(_Z15PrintStackTracev+0x1c)[0xa1898c]
./Cuberite_debug(_ZN9cLuaState4cRef5UnRefEv+0x71)[0xa2f2d1]
./Cuberite_debug(_ZN9cLuaState4cRefD1Ev+0x27)[0xa37257]
./Cuberite_debug[0xa8ace6]
./Cuberite_debug[0xa8ad29]
./Cuberite_debug(_ZNKSt14default_deleteI19cChunkCoordCallbackEclEPS0_+0x2f)[0x901b3f]
./Cuberite_debug(_ZNSt10unique_ptrI19cChunkCoordCallbackSt14default_deleteIS0_EED2Ev+0x43)[0x8fd443]
./Cuberite_debug(_ZN15cLightingThread18cLightingChunkStayD2Ev+0x34)[0x95c614]
./Cuberite_debug(_ZN15cLightingThread18cLightingChunkStayD0Ev+0x19)[0x95c649]
./Cuberite_debug(_ZN15cLightingThread7ExecuteEv+0x20a)[0x95b19a]
./Cuberite_debug(_ZN9cIsThread9DoExecuteEv+0x37)[0xa10ba7]
./Cuberite_debug(_ZSt13__invoke_implIvRKM9cIsThreadFvvEPS0_JEET_St21__invoke_memfun_derefOT0_OT1_DpOT2_+0x7d)[0xa1150d]
./Cuberite_debug(_ZSt8__invokeIRKM9cIsThreadFvvEJPS0_EENSt9result_ofIFOT_DpOT0_EE4typeES8_SB_+0x32)[0xa11482]
./Cuberite_debug(_ZNKSt12_Mem_fn_baseIM9cIsThreadFvvELb1EEclIJPS0_EEEDTclsr3stdE8__invokedtdefpT6_M_pmfspclsr3stdE7forwardIT_Efp_EEEDpOS6_+0x30)[0xa11400]
./Cuberite_debug(_ZNSt12_Bind_simpleIFSt7_Mem_fnIM9cIsThreadFvvEEPS1_EE9_M_invokeIJLm0EEEEvSt12_Index_tupleIJXspT_EEE+0x42)[0xa113a2]
./Cuberite_debug(_ZNSt12_Bind_simpleIFSt7_Mem_fnIM9cIsThreadFvvEEPS1_EEclEv+0x15)[0xa11355]
./Cuberite_debug(_ZNSt6thread11_State_implISt12_Bind_simpleIFSt7_Mem_fnIM9cIsThreadFvvEEPS3_EEE6_M_runEv+0x19)[0xa11119]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xb9e6f)[0x7f9a563dce6f]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x7494)[0x7f9a55ef6494]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f)[0x7f9a55c38aff]
Cuberite_debug: /home/lukas/cpp/cuberite/src/Bindings/LuaState.cpp:2402: void cLuaState::cRef::UnRef(): Assertion `0' failed.
```

**Comment by peterbell10** [comment]
I guess`cLuaState::cRef` shouldn't be used directly if it's expecting a lock.  I'll look for an alternative.

**Comment by peterbell10** [comment]
I changed it to use a `cLuaState::cOptionalCallback` which does the right locking and I couldn't get it to crash on msvc or gcc.

**Comment by Seadragon91** [comment]
Thanks :)

File: src/Bindings/ManualBindings_World.cpp [hunk 1]
```diff
@@ -430,31 +430,23 @@ static int tolua_cWorld_PrepareChunk(lua_State * tolua_S)
 		public cChunkCoordCallback
 	{
 	public:
-		cCallback(lua_State * a_LuaState):
-			m_LuaState(a_LuaState),
-			m_Callback(m_LuaState, 4)
+		cCallback(cLuaState & a_LuaState)
 		{
+			m_Callback.RefStack(a_LuaState, 4);
```
Comment (madmaxoft) [inline src/Bindings/ManualBindings_World.cpp]: Might be better to actually read this callback on line 420, with `GetStackValues`.

**Pull request merged**